src/views/ProductsDetails.vue

```diff
@@ -0,0 +1,212 @@
+          >
+            <template v-slot:append>
+              <v-icon @click="addAuthorHandler">mdi-plus</v-icon>
+            </template>
+          </v-text-field>
+
+          <div>
+            <v-chip
+              v-for="(author, index) in authors"
+              :key="index"
+              closable
+              @click="removeAuthor(index)"
+            >
+              {{ author }}
+            </v-chip>
+          </div>
+
+          <v-text-field
+            class="mt-4"
+            v-model="salePrice"
+            label="Sale price"
+            :rules="useNumberFieldRules()"
+          ></v-text-field>
+
+          <resources-dialog
+            v-model="resourceDialog"
+            @save-resources-dialog="resourcesTableValues"
+            @close-dialog="closeDialog"
+          ></resources-dialog>
+
+          <products-dialog v-model="productsDialog" @close-dialog="closeDialog">
+          </products-dialog>
+
+          <div class="d-flex flex-column">
+            <div class="d-flex justify-space-between">
+              <v-btn color="primary" @click="resourceDialog = true">
+                Resources
+              </v-btn>
+              <v-btn color="primary" @click="productsDialog = true">
+                Products
+              </v-btn>
+            </div>
+
+            <v-btn color="success" class="mt-4" block type="submit">
+              Submit
+            </v-btn>
+
+            <v-btn color="error" class="mt-4" block @click="resetForm">
+              Reset
+            </v-btn>
+
+            <v-btn color="warning" class="mt-4" block @click="router.go(-1)">
+              Go Back
+            </v-btn>
+          </div>
+        </v-form>
+      </v-sheet>
+    </v-card>
+  </v-container>
+</template>
+
+<script setup>
+import {
+  useTextFieldLargeRules,
+  useNumberFieldRules,
+  useTextAreaFieldRules,
+} from "@/utils/validation-rules";
+
+import { ref, computed, inject } from "vue";
+import { useRoute, useRouter } from "vue-router";
+import { useStore } from "vuex";
+import ResourcesDialog from "../components/Dialog/ResourcesDialog.vue";
+import ProductsDialog from "@/components/Dialog/ProductsDialog.vue";
+
+const props = defineProps(["VDataTable"]);
+const store = useStore();
+const route = useRoute();
+const router = useRouter();
+const snackbarProvider = inject("snackbarProvider");
+
+const user = computed(() => store.getters["auth/getUser"]).value;
+
+try {
+  await store.dispatch("users/fetchResourcesForUser", user.id);
+} catch (error) {
+  console.log(error);
+}
+
+const [
+  pageTitle,
+  resourceDialog,
+  productsDialog,
+  form,
+  productName,
+  description,
+  authorInput,
+  authors,
+  salePrice,
+  authorsValidation,
+  resourcesContent,
+  productsContent,
+] = [
+  ref(route.meta.title),
+  ref(false),
+  ref(false),
+  ref(null),
+  ref(""),
+  ref(""),
+  ref(""),
+  ref([]),
+  ref(""),
+  ref(false),
+  ref([]),
+  ref(["g28a891r-df13-40eg-9e51-ce313h52f6fi"]),
+];
```

Comment:
Is this the preferred way? When there are so many fields I can't follow what field is initialized with what value.